sidechain/src/tests.rs

```diff
@@ -202,6 +208,82 @@ fn dont_process_confirmation_of_second_registered_enclave() {
 	})
 }
 
+#[test]
+fn confirm_imported_sidechain_block_works_for_correct_shard_with_updated_fingerprint() {
+	new_test_ext().execute_with(|| {
+		Timestamp::set_timestamp(TEST7_TIMESTAMP);
+		run_to_block(1);
+		let enclave_signer = AccountKeyring::Eve.to_account_id();
+		let enclave =
+			register_sovereign_test_enclave(&enclave_signer, EnclaveFingerprint::default());
+		let shard = ShardIdentifier::from(enclave.fingerprint());
+		let hash = H256::default();
+		// initialize shard
+		assert_ok!(EnclaveBridge::update_shard_config(
+			RuntimeOrigin::signed(enclave_signer.clone()),
+			shard,
+			ShardConfig::new(enclave.fingerprint()),
+			0,
+		));
+
+		assert_ok!(Sidechain::confirm_imported_sidechain_block(
+			RuntimeOrigin::signed(enclave_signer.clone()),
+			shard,
+			1,
+			2,
+			hash
+		));
+
+		let expected_event = RuntimeEvent::Sidechain(SidechainEvent::FinalizedSidechainBlock(
+			enclave_signer.clone(),
+			hash,
+		));
+		assert!(System::events().iter().any(|a| a.event == expected_event));
+
+		let new_fingerprint = EnclaveFingerprint::from([2u8; 32]);
+
+		assert_ok!(EnclaveBridge::update_shard_config(
+			RuntimeOrigin::signed(enclave_signer.clone()),
+			shard,
+			ShardConfig::new(new_fingerprint),
+			1,
+		));
+
+		// should still work with old enclave because update not yet enacted
+		assert_ok!(Sidechain::confirm_imported_sidechain_block(
+			RuntimeOrigin::signed(enclave_signer.clone()),
+			shard,
+			2,
+			3,
+			hash
+		));
+
+		let expected_event = RuntimeEvent::Sidechain(SidechainEvent::FinalizedSidechainBlock(
+			enclave_signer.clone(),
+			hash,
+		));
+		assert!(System::events().iter().any(|a| a.event == expected_event));
+
+		run_to_block(2);
+
+		// enclave upgrade of instance takes place
+		register_sovereign_test_enclave(&enclave_signer, new_fingerprint);
+
+		// now retry as new enclave with same signer
+		assert_ok!(Sidechain::confirm_imported_sidechain_block(
+			RuntimeOrigin::signed(enclave_signer.clone()),
+			shard,
+			3,
+			4,
+			hash
+		));
+
+		let expected_event =
+			RuntimeEvent::Sidechain(SidechainEvent::FinalizedSidechainBlock(enclave_signer, hash));
+		assert!(System::events().iter().any(|a| a.event == expected_event));
+	})
+}
```

Comment:
this here is pretty much the raison-d-être for this PR. This test should demonstrate how enclave upgrades can be done